optaplanner-core/src/test/java/org/optaplanner/core/impl/domain/common/accessor/MemberAccessorFactoryTest.java

```diff
@@ -0,0 +1,147 @@
+import org.optaplanner.core.impl.testdata.domain.reflect.field.TestdataFieldAnnotatedEntity;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertSame;
+import static org.optaplanner.core.impl.testdata.util.PlannerAssert.*;
+
+public class MemberAccessorFactoryTest {
+
+    @Test
+    public void fieldAnnotatedEntity() throws NoSuchFieldException {
+        MemberAccessor memberAccessor = MemberAccessorFactory.buildMemberAccessor(
+                TestdataFieldAnnotatedEntity.class.getDeclaredField("value"),
+                MemberAccessorFactory.MemberAccessorType.FIELD_OR_GETTER_METHOD_WITH_SETTER, PlanningVariable.class);
+        assertEquals("value", memberAccessor.getName());
+        assertEquals(TestdataValue.class, memberAccessor.getType());
+
+        TestdataValue v1 = new TestdataValue("v1");
+        TestdataValue v2 = new TestdataValue("v2");
+        TestdataFieldAnnotatedEntity e1 = new TestdataFieldAnnotatedEntity("e1", v1);
+        assertSame(v1, memberAccessor.executeGetter(e1));
+        memberAccessor.executeSetter(e1, v2);
+        assertSame(v2, e1.getValue());
+    }
+
+
+    @Test
+    public void privateField() throws NoSuchFieldException {
+        MemberAccessor memberAccessor = MemberAccessorFactory.buildMemberAccessor(
+                TestdataVisibilityModifierSolution.class.getDeclaredField("privateField"),
+                MemberAccessorFactory.MemberAccessorType.FIELD_OR_GETTER_METHOD_WITH_SETTER, ProblemFactProperty.class);
+        assertEquals("privateField", memberAccessor.getName());
+        assertEquals(String.class, memberAccessor.getType());
+
+        TestdataVisibilityModifierSolution s1 = new TestdataVisibilityModifierSolution("s1",
+                "firstValue", "n/a",
+                "n/a", "n/a", "n/a", "n/a");
+        assertEquals("firstValue", memberAccessor.executeGetter(s1));
+        memberAccessor.executeSetter(s1, "secondValue");
+        assertEquals("secondValue", memberAccessor.executeGetter(s1));
+    }
+
+    @Test
+    public void publicField() throws NoSuchFieldException {
+        MemberAccessor memberAccessor = MemberAccessorFactory.buildMemberAccessor(
+                TestdataVisibilityModifierSolution.class.getDeclaredField("publicField"),
+                MemberAccessorFactory.MemberAccessorType.FIELD_OR_GETTER_METHOD_WITH_SETTER, ProblemFactProperty.class);
+        assertEquals("publicField", memberAccessor.getName());
+        assertEquals(String.class, memberAccessor.getType());
+
+        TestdataVisibilityModifierSolution s1 = new TestdataVisibilityModifierSolution("s1",
+                "n/a", "firstValue",
+                "n/a", "n/a", "n/a", "n/a");
+        assertEquals("firstValue", memberAccessor.executeGetter(s1));
+        memberAccessor.executeSetter(s1, "secondValue");
+        assertEquals("secondValue", memberAccessor.executeGetter(s1));
+    }
+
+    @Test @Ignore("Non-public getters/setters are not supported")
```

Comment:
@ge0ffrey Why not remove the test instead of ignoring it? Is there any chance the test will be unignored in the future? I guess it won't be.

Reply:
Done, thanks